packages/modelserver-theia/src/browser/model-server-frontend-contribution.ts

```diff
@@ -26,12 +27,27 @@ export class ModelServerFrontendContribution implements FrontendApplicationContr
     }
 
     async setup(): Promise<void> {
-        this.workspaceService.onWorkspaceChanged(workspace => {
+        this.workspaceService.onWorkspaceChanged(async workspace => {
             if (workspace[0] && workspace[0].resource) {
                 const workspaceRoot = new URI(workspace[0].resource.toString());
                 const uiSchemaFolder = workspaceRoot.clone().segment('.ui-schemas');
-                this.modelServerClient.configureServer({ workspaceRoot, uiSchemaFolder });
+                await this.waitForReady();
+                console.log('Model Server ready');
+                await this.modelServerClient.configureServer({ workspaceRoot, uiSchemaFolder });
             }
         });
     }
+
+    async waitForReady(ms = 1000): Promise<void> {
+        let available = await this.modelServerClient.ping().catch(e => {
```

Comment:
why not start with `available=false` and revert the order in the while loop to have the ping only once in the method?

Reply:
Then we would wait an additional second after the server was pinged successfully, right?

So we probably need:

```ts
    async waitForReady(ms = 1000): Promise<void> {
        let available = false;
        while (!available) {
            await this.modelServerClient
                .ping()
                .then(async value => {
                    if (value) {
                        available = true;
                    } else {
                        await timeout(ms);
                    }
                })
                .catch(async e => {
                    console.log('Model Server not ready yet: ' + e);
                    await timeout(ms);
                });
        }
        return;
    }
```

WDYT?